content/docs/connect/connection-pooling.md

```diff
@@ -60,12 +60,22 @@ Even with the largest compute size, the `max_connections` limit may not be suffi
+
+To ensure that direct access to Postgres is still possible for e.g. administrative tasks, the pooler is configured to only open up to [64 connections](#neon-pgbouncer-configuration-settings) to Postgres for each user to each database. I.e., there can be only 64 active connections by `john` to the `neondb` database through the pooler. All other connections by `john` to the `neondb` database will have to wait for one of those 64 active connections to complete their transactions before the next connection's work is started.  
+At the same time, a user `mike` will also be able to connect to the `neondb` database through the pooler and have up to 64 concurrent active transactions across 64 connections, assuming the endpoint started with a high enough `minCU` setting to be configured with a high enough `max_connections` setting to support those 128 concurrent connections from those two users.  
+Similarly, even if the user `john` has 64 concurrently active transactions through the pooler to the `neondb` database, that user can still start up to 64 concurrent transactions in the `john_db` database when connected through the pooler; but again, only if Postgres' `max_connections` limit has the capacity for the connections that are managed by the pooler.
+
+For further information, see [PgBouncer](#pgbouncer).
 
 <Admonition type="important">
-Connection pooling supports up to 10,000 concurrent connections when multiple Postgres users are connecting to multiple Postgres databases. For a single user connecting to a single database, the maximum number of concurrent connections is 64. This limit is controlled by the PgBouncer `default_pool_size` parameter, which defines the default number of server connections per user/database pair. See [Neon PgBouncer configuration settings](#neon-pgbouncer-configuration-settings). If your application connects to your database using a single Postgres role, expect to encounter this limit when exceeding the 64 concurrent connection limit. For information about creating additional Postgres roles, see [Manage roles](/docs/manage/roles#manage-roles-with-sql). Roles can created via the Neon Console, CLI, API, and SQL.
+You will not be able to get interactive results from all 10,000 connections at the same time. Connections to the pooler endpoint still consume some connections on the main endpoint: PgBouncer forwards operations from the user's connections through its own pool of connnections to Postgres, and adaptively adds more connections to PostgreSQL as and when needed by more concurrently active user connections. The 10,000 connection limit is therefore most useful for "serverless" applications and application-side connection pools that have many open connections, but infrequent and/or short transactions.  
```

Comment:
> but infrequent and/or short transactions

- what's a "short" transaction?
- each connection that executes a query, and takes greater than `query_wait_timeout` time is a "long" transaction?

Reply:
> That depends on the application, but generally I'd transactions short when the duration of the whole transaction is within one order of magnitude of the RTT to the application.

Shall we update the wording? Because this gets caught in a loop. So a hypothetical case that's bound to come:

- a user assumes/thinks that their transaction is short
- they report
- they'd be let known that their transaction was instead long
- they come back - ask for what's a long transaction?
- well, that depends and is specific to your case, which in this case it was it

Reply:
> I'd say so, yes, but that depends on query_wait_timeout. You can configure that setting down to 1ms without issues (other than that most significant queries will fail at that point).

Let's definitely including the `query_wait_timeout` as the limit for long or short transaction?

> You can configure that setting down to 1ms without issues (other than that most significant queries will fail at that point).

Would we want to document this?

Reply:
Let's keep the changes as proposed and for "short" transactions let's just link to our hosted postgres docs so people can read more if need be: https://neon.tech/docs/postgres/tutorial-transactions .